crate/client/src/config.rs

```diff
@@ -256,15 +256,26 @@ impl ClientConf {
         Ok(conf)
     }
 
-    pub fn initialize_kms_client(&self) -> Result<KmsClient, ClientError> {
+    /// Initialize a KMS REST client.
+    ///
+    /// Parameters `kms_server_url` and `accept_invalid_certs` from the command line
+    /// will override the ones from the configuration file.
+    pub fn initialize_kms_client(
+        &self,
+        kms_server_url: Option<&str>,
+        accept_invalid_certs: Option<bool>,
+    ) -> Result<KmsClient, ClientError> {
+        let kms_server_url = kms_server_url.unwrap_or(&self.kms_server_url);
+        let accept_invalid_certs = accept_invalid_certs.unwrap_or(self.accept_invalid_certs);
+
         // Instantiate a KMS server REST client with the given configuration
         let kms_rest_client = KmsClient::instantiate(
-            &self.kms_server_url,
+            kms_server_url,
             self.kms_access_token.as_deref(),
             self.ssl_client_pkcs12_path.as_deref(),
             self.ssl_client_pkcs12_password.as_deref(),
             self.kms_database_secret.as_deref(),
-            self.accept_invalid_certs,
+            accept_invalid_certs,
             if let Some(certificate) = &self.verified_cert {
                 Some(Certificate(
                     X509Certificate::from_pem(certificate.as_bytes())?.to_der()?,
```

Comment:
if this returns error, the in the upper scope might panic, as the result is often unwrapped.

Reply:
It is true that so far I see the risk of panicking only in test and benches. Idk if it is safe to panic there.

Reply:
the `unwrap_or` does not panic. It returns the contained [Some](vscode-file://vscode-app/usr/share/code/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) value or a provided default.